tests/validators/answers/test_number_answer_validator.py

```diff
@@ -80,6 +80,118 @@ def test_are_decimal_places_valid():
     }
 
 
+def test_maximum_set_as_string():
+    answer = {
+        "id": "total-percentage",
+        "label": "Total",
+        "mandatory": False,
+        "q_code": "10002",
+        "type": "Percentage",
+        "maximum": {"value": "100"},
+    }
+
+    validator = NumberAnswerValidator(
+        answer, get_mock_schema_with_data_version("0.0.3")
+    )
+    validator.validate_min_max_is_number()
+
+    assert validator.errors[0] == {
+        "message": validator.MAX_MIN_IS_STRING,
+        "answer_id": "total-percentage",
+    }
+
+
+def test_minimum_set_as_string():
+    answer = {
+        "calculated": True,
+        "description": "The total percentages should be 100%",
+        "id": "total-percentage",
+        "label": "Total",
+        "mandatory": False,
+        "q_code": "10002",
+        "type": "Percentage",
+        "minimum": {"value": "0"},
+    }
+
+    validator = NumberAnswerValidator(
+        answer, get_mock_schema_with_data_version("0.0.3")
+    )
+    validator.validate_min_max_is_number()
+
+    assert validator.errors[0] == {
+        "message": validator.MAX_MIN_IS_STRING,
+        "answer_id": "total-percentage",
+    }
+
+
+def test_min_and_max_set_as_string():
+    answer = {
+        "calculated": True,
+        "description": "The total percentages should be 100%",
+        "id": "total-percentage",
+        "label": "Total",
+        "mandatory": False,
+        "q_code": "10002",
+        "type": "Percentage",
+        "maximum": {"value": "100"},
+        "minimum": {"value": "0"},
+    }
+
+    validator = NumberAnswerValidator(
+        answer, get_mock_schema_with_data_version("0.0.3")
+    )
+    validator.validate_min_max_is_number()
+
+    assert validator.errors[1] == {
+        "message": validator.MAX_MIN_IS_STRING,
+        "answer_id": "total-percentage",
+    }
+    assert len(validator.errors) == 2
+
+
+def test_min_if_set_as_float():
+    answer = {
+        "id": "answer-2",
+        "mandatory": True,
+        "type": "Currency",
+        "label": "Money spent on vegetables",
+        "description": "Enter the full value",
+        "minimum": {"value": 0.00, "exclusive": True},
+    }
+
+    validator = NumberAnswerValidator(
+        answer, get_mock_schema_with_data_version("0.0.3")
+    )
+
+    validator.validate_min_max_is_number()
+
+    assert len(validator.errors) == 0
+
+
+def test_max_if_not_set_as_integer():
+    answer = {
+        "id": "set-maximum-answer",
+        "description": "This is a description of the maximum value",
+        "label": "Set a value less than the total above",
+        "mandatory": True,
+        "type": "Currency",
+        "currency": "GBP",
+        "decimal_places": 2,
+        "maximum": {
+            "value": {
+                "source": "calculated_summary",
+                "identifier": "currency-total-playback",
```

Comment:
This also only passes for the reason above where value is a dictionary, I think you'll need to use a real schema for this test so you can correctly determine that the value source is of a numeric type